river/neighbors/__init__.py

```diff
@@ -4,9 +4,15 @@
 until a query is received.
 
 """
-from .knn_adwin import KNNADWINClassifier
+from .base_neighbors import BaseKNN
 from .knn_classifier import KNNClassifier
 from .knn_regressor import KNNRegressor
-from .sam_knn import SAMKNNClassifier
+from .neighbors import MinkowskiNeighbors, NearestNeighbors
 
-__all__ = ["KNNADWINClassifier", "KNNClassifier", "KNNRegressor", "SAMKNNClassifier"]
+__all__ = [
+    "BaseKNN",
+    "MinkowskiNeighbors",
```

Comment:
I don't think we want to make these classes public, right?

Reply:
I think baseKNN yes, that's the one I want to use, MinkowskiNeighbors probably not, at least until someone complains and says they want to use it and don't understand why they can't. I am more inclined to be conservative and expose both.

Reply:
Don't you mean conservative and not expose both? I think the less we expose the better, no? I'm not sure I follow

Reply:
I mean conservative in the "develop so users are less likely to open issues on something" but I'm good with removing those from being exposed - see most recent commit!